### PR TITLE
Add a setting to specify default repositories

### PR DESCRIPTION
**Pull request by jmcphers**
This change adds a new setting that allows end users to select the default repository used in R sessions. Prior to the change, we always used `cran.rstudio.com` if no other repository was set; this remains the default, but now there are several other choices:

image

For compatibility with shared RStudio configurations, the `auto` setting also checks for a `repos.conf` file in well-known XDG locations. You can read more about the expected format of this file here: https://solutions.posit.co/envs-pkgs/rsw_defaults/#repos.conf

Addresses https://github.com/posit-dev/positron/issues/5509

> [!NOTE]
>
> This change depends on a change from `ark` (https://github.com/posit-dev/ark/pull/645) and should not be merged until that change is.

### QA Notes

- These repositories are _defaults_ and shouldn't override anything you've already set in e.g. `.Rprofile` or `Rprofile.site`.
- It is not strictly necessary to restart Positron to apply changes to the new option; you only need to shut down your R session (not restart it) and create a new one. We say "restart Positron" because it's easier for people to understand.
- The XDG locations searched include both Positron and RStudio directories. On macOS, for example, `repos.conf` can be placed in `~/Library/Preferences/positron/repos.conf`.

### CODE REVIEW

File: extensions/positron-r/src/kernel-spec.ts [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+	const profile = config.get<string>('kernel.profile');
+
+
+	/* eslint-disable */
+	const env = <Record<string, string>>{
+		'RUST_BACKTRACE': '1',
+		'RUST_LOG': logLevelForeign + ',ark=' + logLevel,
+		'R_HOME': rHomePath,
+		...userEnv
+	};
```
Comment (jennybc) [inline extensions/positron-r/src/kernel-spec.ts:51]: Minor observation: we are using a little interface for env vars elsewhere in positron-r (but I don't think this matters that much and I see this code pre-existed, it's just moving in this PR):

https://github.com/posit-dev/positron/blob/1492e58b74678103a84810ae453cfa65cc4662ad/extensions/positron-r/src/session.ts#L30-L32

https://github.com/posit-dev/positron/blob/1492e58b74678103a84810ae453cfa65cc4662ad/extensions/positron-r/src/uri-handler.ts#L69

https://github.com/posit-dev/positron/blob/1492e58b74678103a84810ae453cfa65cc4662ad/extensions/positron-r/src/session.ts#L927

File: extensions/positron-r/package.nls.json [hunk 2]
```diff
@@ -59,5 +59,10 @@
 	"r.configuration.pipe.native.description": "Native pipe available in R >= 4.1",
 	"r.configuration.pipe.magrittr.description": "Pipe operator from the magrittr package, re-exported by many other packages",
 	"r.configuration.diagnostics.enable.description": "Enable R diagnostics globally",
-	"r.configuration.taskHyperlinks.description": "Turn on experimental support for hyperlinks in package development tasks"
+	"r.configuration.taskHyperlinks.description": "Turn on experimental support for hyperlinks in package development tasks",
+	"r.configuration.defaultRepositories.description": "The default repositories to use for R package installation, if no repository is otherwise specified in R startup scripts (restart Positron to apply).\n\nThe default repositories will be set as the `repos` option in R.",
+	"r.configuration.defaultRepositories.auto.description": "Automatically choose a default repository, or use a repos.conf file if it exists.",
+	"r.configuration.defaultRepositories.rstudio.description": "Use the RStudio CRAN mirror (cran.rstudio.com)",
```
Comment (jennybc) [inline extensions/positron-r/package.nls.json:65]: Taking this chance to have a side discussion:

In my `.Rprofile`, I have

```
...
repos = c(
    P3M = "https://p3m.dev/cran/latest",
    CRAN = "https://cloud.r-project.org"
)
...
```

In terms of what we feature, is there a reason to push cran.rstudio.com over cloud.r-project.org? I guess I've implicitly preferred the latter from some super vague reasons around rstudio->posit rebrand and avoiding overt branding.

**Review by jennybc — COMMENTED** [review]
Leaving two comments this PR inspired. About to try this out locally.

**Review by jennybc — DISMISSED** [review]
I played around with this, with an appropriate ark version, and all seems well. I can switch between the different enum values and I can also use a `repos.conf` file with "auto".

**Review by jennybc — APPROVED** [review]

File: extensions/positron-r/src/kernel-spec.ts (hunk 1 above)
Comment (jmcphers) [reply extensions/positron-r/src/kernel-spec.ts:51]: Good observation! Indeed I didn't touch this (just moved it to keep the file from becoming too unwieldy) but that'd be an improvement.

File: extensions/positron-r/package.nls.json (hunk 2 above)
Comment (jmcphers) [reply extensions/positron-r/package.nls.json:65]: I don't think we necessarily want to push cran.rstudio.com (Posit also sponsors `cloud.r-project.org`), I just used it to match ark's existing behavior. Might be worth providing a `cloud.r-project.org` option too, let's see what folks say.

**Pull request merged**